src/isotools/_transcriptome_io.py

```diff
@@ -396,7 +414,7 @@ def add_sample_from_bam(self, fn, sample_name=None, barcode_file=None, fuzzy_jun
                             if save_readnames:
                                 tr['reads'].append(read.query_name)
                             break
-                    else:
+                    else: # if not (for-else block), create a new transcript (tr) entry and add it to the transcripts data structure
```

Comment:
And this one in the line below the `else`
Inline comments are generally discouraged, especially if they are long